docs/user-guide/reminders-and-external-events.rst

```diff
@@ -0,0 +1,257 @@
+    """Schedules a reminder, supplied with the last message's entities."""
+
+    def name(self) -> Text:
+        return "action_set_reminder"
+
+    def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+
+        dispatcher.utter_message("I will remind you in 5 seconds.")
+
+        date = datetime.datetime.now() + datetime.timedelta(seconds=5)
+        entities = tracker.latest_message.get("entities")
+
+        reminder = ReminderScheduled(
+            "EXTERNAL_reminder",
+            trigger_date_time=date,
+            entities=entities,
+            name="my_reminder",
+            kill_on_user_message=False,
+        )
+
+        return [reminder]
+
+
+Note, that this requires the ``datetime`` and ``rasa-sdk.events`` packages.
+For details, have a look at the ``reminderbot`` example under ``rasa/examples/reminderbot``.
+
+Finally, we define another custom action ``action_react_to_reminder`` and link it to the ``EXTERNAL_reminder`` intent:
+
+.. code-block:: md
+
+  - EXTERNAL_reminder:
+    triggers: action_react_to_reminder
+
+where the ``action_react_to_reminder`` is
+
+.. code-block:: python
+
+  class ActionReactToReminder(Action):
+    """Reminds the user to call someone."""
+
+    def name(self) -> Text:
+        return "action_react_to_reminder"
+
+    def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+
+        name = next(tracker.get_latest_entity_values("name"), None) or "someone"
+        dispatcher.utter_message(f"Remember to call {name}!")
+
+        return []
+
+Instead of a custom action, we could also have used a simple response template.
+But here we want to make use of the fact that the reminder can carry entities, and we can process the entities in this custom action.
+
+.. warning::
+
+  Reminders are cancelled whenever you shutdown rasa.
+
+
+.. warning::
+
+  Reminders currently (Rasa 1.7) don't work in `rasa shell`.
+  Use `rasa x` instead.
+
+Check out the ``reminderbot`` example project under ``rasa/examples/reminderbot``, and feel free to customize things for your own assistant!
+
+
+.. _cancelling-reminders-guide:
+
+Cancelling reminders
+^^^^^^^^^^^^^^^^^^^^
+
+Sometimes the user may want to cancel a reminder that he has scheduled earlier.
+A simple way of adding this functionality to your assistant is to create an intent ``ask_forget_reminders`` and let your assistant respond to it with a custom action such as
+
+.. code-block:: python
+
+  class ForgetReminders(Action):
+    """Cancels all reminders."""
+
+    def name(self) -> Text:
+        return "action_forget_reminders"
+
+    def run(
+        self, dispatcher, tracker: Tracker, domain: Dict[Text, Any]
+    ) -> List[Dict[Text, Any]]:
+
+        # Cancel all reminders
+        return [ReminderCancelled()]
```

Comment:
What do you think about adding feeback? e.g. "Okay, I'll cancel all your reminders."